cp2k: add new version and tweak settings

glennpj opened the pull request:
The @2022: versions use different settings than before when using CUDA.

glennpj committed:
cp2k: add new version and tweak settings

The @2022: versions use different settings than before when using CUDA.

alalazo reviewed (dismissed):
Waiting for @dev-zero to :heavy_check_mark: before merging

dev-zero commented on var/spack/repos/builtin/packages/cp2k/package.py:
```diff
@@ -577,6 +587,10 @@ def edit(self, spec, prefix):
+
+                if spec.satisfies("+cuda_fft"):
+                    cppflags += ["-D__PW_CUDA"]
+                    libs += ["-lcufft"]
```
you disabled the availability of the `cuda_fft` variant above I think

alalazo replied:
From the perspective of Spack directives, this condition branch is entered only for version `9.*` when `+cuda_fft`. Not sure if this makes sense for `cp2k` though.

dev-zero replied:
@alalazo exactly what I meant. In this PR there is also a `when="@:9",  # req in CP2K v2022+` added for the `cuda_fft` variant.

glennpj replied:
I do not think `cuda_fft` is disabled above as it is enabled by default in `@2022:`. That is why I set the condition on the variant. However, looking at the cp2k docs again, I see that it _can_ be disabled in the 2022 version. If I am understanding correctly, versions up to 9 would use `D__PW_CUDA` to turn it on, and versions 2022 and above would use `D__NO_OFFLOAD_PW` to turn it off. So, I should redo the logic to reflect that. Does that sound correct @dev-zero?

glennpj replied:
I added a commit to enable the `cuda_fft` variant for `@2022:` and adjusted the `if...` logic.

dev-zero replied:
I think that looks right, yes.
Thanks a lot!

glennpj committed:
Restore cuda_fft variant

The default for cuda_fft changed from OFF to ON in @2022:. I did not
initially see the new variable that allowed turning it OFF, thus setting
a version condition on the variant. The cuda_fft variant has been
enabled again for @2022: and the if... logic adjusted accordingly.

alalazo commented:
@spackbot run pipeline
